datashader/utils.py

```diff
@@ -383,3 +383,74 @@ def dataframe_from_multiple_sequences(x_values, y_values):
    return pd.DataFrame({'x': x, 'y': y.flatten()})
 
    
+def pd_mesh(vertices, simplices):
+    """Merge vertices and simplices into a triangular mesh, suitable to be
+    passed into the ``Canvas.trimesh()`` method via the ``mesh``
+    keyword-argument. Both arguments are assumed to be Pandas DataFrame
+    objects.
+    """
+    # Verify the simplex data structure
+    assert simplices.values.shape[1] >= 3, ('At least three vertex columns '
+                                            'are required for the triangle '
+                                            'definition')
+    simplices_all_ints = simplices.dtypes.iloc[:3].map(
+        lambda dt: np.issubdtype(dt, np.integer)
+    ).all()
+    assert simplices_all_ints, ('Simplices must be integral. You may '
+                                'consider casting simplices to integers '
+                                'with ".astype(int)"')
+
```

Comment:
This validation code could then move into mesh() to avoid duplication.

Reply:
I thought about putting the validation code in one place to avoid duplication, however I kept them separate due to the subtle differences between the dask and pandas APIs. They ended up being the same in this case, but the xarray version will almost certainly be different. I'll default to your judgement on this one, though.

Reply:
No need to preemptively add complexity; just make a special case for xarray if and when it is needed.  So please do combine them for now.  I find duplicating error messages to be especially confusing in practice, because it's then not possible to track down which bit of code generated the message.  Much better for all messages to have one point of orgin...

Reply:
Thanks for the update!  The same issue applies at the end of each function, too, though; the entire bit after "# If vertices don't have weights..." is the same in both and contains an error message, so please move that into the same function to make the differences between the two versions clearer, to avoid code duplication, and to give a single generation point for each error message.

Reply:
Damn, I see the ".compute()" now.  Any obvious way around duplicating that, then?

Reply:
It would be nice if pandas made .compute() into a no-op to allow compatibility!

Reply:
Just pushed up the change for de-duplication.

Reply:
Best I can see is to write a trivial helper function like:

```
def compute(obj,dask=False):
    return obj.compute() if dask else obj
```

Reply:
I'm planning to add dask support in another PR, in a way that hopefully won't require us to use `.compute()` at all; I'm only using it here as a stopgap until I get the chance to make the code more dask-friendly.

Reply:
Ok.